### PR TITLE
Show messagebox when dronebl error occurs

### PR DESCRIPTION
**Pull request by GrantBartlett**
ErrorMessage

### CODE REVIEW

File: DXMainClient/DXGUI/Generic/TopBar.cs [hunk 1]
```diff
@@ -224,10 +224,17 @@ public override void Initialize()
...
             connectionManager.ConnectAttemptFailed += ConnectionManager_ConnectAttemptFailed;
+            connectionManager.ErrorDroneBL += ConnectionManager_ErrorDroneBL;
 
             privateMessageHandler.UnreadMessageCountUpdated += PrivateMessageHandler_UnreadMessageCountUpdated;
         }
 
+        private void ConnectionManager_ErrorDroneBL(object sender, string message)
+        {
+            XNAMessageBox messageBox = new XNAMessageBox(WindowManager, "Error connecting", message, XNAMessageBoxButtons.OK);
+            messageBox.Show();
```
Comment (devo1929) [inline DXMainClient/DXGUI/Generic/TopBar.cs:235]: You should be able to just do:
`XNAMessageBox.Show(WindowManager, "Error connecting", message);`

File: DXMainClient/Online/CnCNetManager.cs [hunk 2]
```diff
@@ -519,6 +520,11 @@ public void OnGenericServerMessageReceived(string message)
 
         private void DoGenericServerMessageReceived(string message)
         {
+            if (message.Contains("DroneBL"))
```
Comment (devo1929) [inline DXMainClient/Online/CnCNetManager.cs:523]: I can't remember what calls `DoGenericServerMessageReceived`. This won't get triggered by any random person sending the message "DroneBL", right?

**Review by devo1929 — COMMENTED** [review]
left some inline comments

File: DXMainClient/Online/CnCNetManager.cs (hunk 2 above)
Comment (GrantBartlett) [reply DXMainClient/Online/CnCNetManager.cs:523]: I did just check to make sure, it doesn't no. Its server messages received from GameSurge. 

```
08.09. 21:28:44.599    DoGenericServerMessageReceived **  AUTH *** Looking up your hostname 
08.09. 21:28:44.599    DoGenericServerMessageReceived **  AUTH *** Checking Ident 
08.09. 21:28:44.599    DoGenericServerMessageReceived **  AUTH *** Found your hostname 
08.09. 21:28:52.570    DoGenericServerMessageReceived **  AUTH *** No ident response 
08.09. 21:28:52.639    DoGenericServerMessageReceived ** TAL.DE.EU.GameSurge.net:  Your host is TAL.DE.EU.GameSurge.net, running version u2.10.12.18(gs2)
08.09. 21:28:52.639    DoGenericServerMessageReceived ** TAL.DE.EU.GameSurge.net:  This server was created Fri Nov 10 2017 at 11:10:57 CET
08.09. 21:28:52.640    DoGenericServerMessageReceived ** TAL.DE.EU.GameSurge.net:  There are 10 users and 1962 invisible on 13 servers
08.09. 21:28:52.640    DoGenericServerMessageReceived ** TAL.DE.EU.GameSurge.net: 27 operator(s) online
08.09. 21:28:52.640    DoGenericServerMessageReceived ** TAL.DE.EU.GameSurge.net: 1587 channels formed
08.09. 21:28:52.722    DoGenericServerMessageReceived ** TAL.DE.EU.GameSurge.net:  I have 1 clients and 1 servers
08.09. 21:28:52.723    DoGenericServerMessageReceived ** TAL.DE.EU.GameSurge.net Grant Highest connection count: 2 (1 clients) 
08.09. 21:28:52.723    DoGenericServerMessageReceived ** TAL.DE.EU.GameSurge.net Grant on 2 ca 1(10) ft 10(10) tr 
````

**Review by devo1929 — APPROVED** [review]

File: DXMainClient/DXGUI/Generic/TopBar.cs [hunk 3]
```diff
@@ -224,10 +230,37 @@ public override void Initialize()
...
+            {
+                YesClickedAction = OnOpenDroneBLURLClicked
+            };
+            messageBox.Show();
+        }
+
+        private void OnOpenDroneBLURLClicked(XNAMessageBox messageBox)
+        {
+            Process.Start(droneBLErrorLookupURL);
+        }
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Generic/TopBar.cs:262]:
```suggestion
        private void OnOpenDroneBLURLClicked(XNAMessageBox messageBox)
            => Process.Start(droneBLErrorLookupURL);
```

File: DXMainClient/Online/CnCNetManager.cs [hunk 4]
```diff
@@ -519,6 +520,11 @@ public void OnGenericServerMessageReceived(string message)
...
+            {
+                ErrorDroneBL.Invoke(this, new DroneBLErrorEventArgs(message));
+            }
```
Comment (Metadorius) [inline DXMainClient/Online/CnCNetManager.cs:526]:
```suggestion
                ErrorDroneBL.Invoke(this, new DroneBLErrorEventArgs(message));
```

File: DXMainClient/DXGUI/Generic/TopBar.cs (hunk 1 above)
Comment (Metadorius) [reply DXMainClient/DXGUI/Generic/TopBar.cs:256]: Agreed, should do this instead.

**Review by Metadorius — CHANGES_REQUESTED** [review]

File: DXMainClient/DXGUI/Generic/TopBar.cs (hunk 3 above)
Comment (Rans4ckeR) [inline DXMainClient/DXGUI/Generic/TopBar.cs:262]:
```suggestion
        private void OnOpenDroneBLURLClicked(XNAMessageBox messageBox)
	            => ProcessLauncher.StartShellProcess(droneBLErrorLookupURL);
```

**Review by Rans4ckeR — CHANGES_REQUESTED** [review]

File: DXMainClient/DXGUI/Generic/TopBar.cs (hunk 3 above)
Comment (Rans4ckeR) [reply DXMainClient/DXGUI/Generic/TopBar.cs:262]: > 

This is needed for .NET7 or it won't launch the browser.

**Pull request closed without merging**